mitodl/ocw-course-hugo-theme
# refactor webpack_url.html to site_root_url.html and use it on the course image

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-course-hugo-theme/issues/16

#### What's this PR do?
This PR refactors `webpack_url.html` to be named `site_root_url.html` because that name better describes what it does.  Regardless of the `baseUrl` argument being passed into Hugo, URL's passed into this partial will always render relative to the site root.  The course image URL was being rendered using the Hugo `relUrl` function, so instead it's being passed to `site_root_url.html`

#### How should this be manually tested?
 - Generate content using `ocw-to-hugo` and also pass in the arguments `--strips3 --staticPrefix /coursemedia`
 - Verify that in the output's `data` directory, course data templates have a `course_image_url` that looks something like:
```
"course_image_url": "/coursemedia/1-00-introduction-to-computers-and-engineering-problem-solving-spring-2012/457598d84426c61f83ab06ad2aa39c04_1-00s12.jpg",
```
 - In `ocw-course-hugo-starter`, place the following in your `go.mod` file:
```
module github.com/mitodl/ocw-course-hugo-starter

go 1.13

require github.com/mitodl/ocw-course-hugo-theme v0.0.0-20210119222443-12cc8dd8e246 // indirect

```
 - In `ocw-course-hugo-starter`, run:
```
./build.sh -o private/output -c ocw-to-hugo/private/output -b https://ocwnext-rc.odl.mit.edu/courses
```
 - Verify in the output that the course image URL looks like `https://ocwnext-rc.odl.mit.edu/coursemedia/11-438-economic-development-planning-spring-2020/c468a102c494455f0d2d258bebaf0356_11-438f19.jpg` and not like `/courses/11-438-economic-development-planning-spring-2020/coursemedia/11-438-economic-development-planning-spring-2020/c468a102c494455f0d2d258bebaf0356_11-438f19.jpg`
